pulsar-broker/src/test/java/org/apache/pulsar/broker/auth/MockedPulsarServiceBaseTest.java

```diff
@@ -237,9 +237,14 @@ protected final void internalSetupForStatsTest() throws Exception {
     protected void doInitConf() throws Exception {
         this.conf.setBrokerShutdownTimeoutMs(0L);
         this.conf.setLoadBalancerOverrideBrokerNicSpeedGbps(Optional.of(1.0d));
-        this.conf.setBrokerServicePort(Optional.of(0));
+        if (!conf.getBrokerServicePort().isPresent()) {
+            this.conf.setBrokerServicePort(Optional.of(0));
+        }
+
         this.conf.setAdvertisedAddress("localhost");
-        this.conf.setWebServicePort(Optional.of(0));
+        if (!conf.getWebServicePort().isPresent()) {
+            this.conf.setWebServicePort(Optional.of(0));
+        }
```

Comment:
it's better to override `doInitConf` method in a class that needs to override the defaults. please search for examples in the code base. It wouldn't be consistent to make this type of conditions only for these settings.

Reply:
Updated.